android/src/main/kotlin/com/fluttercandies/photo_manager/permission/PermissionDelegate.kt

```
@@ -0,0 +1,175 @@
+     */
+    abstract fun havePermissions(context: Context, requestType: Int): Boolean
+
+    /**
+     * Check if the [mediaLocation] is granted for the user and it is in the manifest.
+     */
+    abstract fun haveMediaLocation(context: Context): Boolean
+
+    companion object {
+        const val requestCode = 3001
+        const val limitedRequestCode = 3002
+
+        /**
+         * Create a [PermissionDelegate] by the sdk version.
+         */
+        fun create(): PermissionDelegate {
+            return when (Build.VERSION.SDK_INT) {
+                in 1 until 23 -> PermissionDelegate19()
+                in 23 until 29 -> PermissionDelegate23()
+                29 -> PermissionDelegate29()
+                in 30 until 33 -> PermissionDelegate30()
+                33 -> PermissionDelegate33()
+                in 34 until Int.MAX_VALUE -> PermissionDelegate34()
+                else -> throw UnsupportedOperationException(
+                    "This sdk version is not supported yet."
+                )
+            }
+        }
+    }
+
+    open fun isHandlePermissionResult(): Boolean {
+        return false
+    }
+
+    open fun handlePermissionResult(
+        permissionsUtils: PermissionsUtils,
+        context: Context,
+        permissions: Array<String>,
+        grantResults: IntArray,
+        needToRequestPermissionsList: MutableList<String>,
+        deniedPermissionsList: MutableList<String>,
+        grantedPermissionsList: MutableList<String>,
+        requestCode: Int
+    ) {
+        throw UnsupportedOperationException(
+            "handlePermissionResult is not implemented," +
+                    " please implement it in your delegate."
+        )
+    }
+
+    open fun presentLimited(
+        permissionsUtils: PermissionsUtils,
+        context: Application,
+        type: Int,
+        resultHandler: ResultHandler
+    ) {
+        LogUtils.debug("[$tag] presentLimited is not implemented")
+        resultHandler.reply(null)
```

Should we reply error here?

---

Currently, there is no possibility of reporting an error when applying for this method.
Additionally, we cannot tell whether the user has selected a new asset.
I don't know under what circumstances I should report an error.